File: crates/wg-ui/src/api/devices.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use wg_shared::types::Device;
use super::ApiResult;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DeviceListResponse {
    pub items: Vec<Device>,
    pub total: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DeviceStatusResponse {
    pub device_id:    Uuid,
    pub connected:    bool,
    pub last_seen_at: Option<i64>,
}

/// GET /api/v1/devices
pub async fn list() -> ApiResult<DeviceListResponse> {
    super::get("/api/v1/devices").await
}

/// GET /api/v1/devices/{id}
pub async fn get(id: Uuid) -> ApiResult<Device> {
    super::get(&format!("/api/v1/devices/{id}")).await
}

/// GET /api/v1/devices/{id}/status
pub async fn status(id: Uuid) -> ApiResult<DeviceStatusResponse> {
    super::get(&format!("/api/v1/devices/{id}/status")).await
}

File: ui/src/api/devices.ts
import { get } from './client'
import type { PaginatedResponse } from '@/types/api'
import type { Device, DeviceStatus, HttpService } from '@/types/device'

export const listDevices = () =>
  get<PaginatedResponse<Device>>('/api/v1/devices')

export const getDevice = (id: string) =>
  get<Device>(`/api/v1/devices/${id}`)

export const getDeviceStatus = (id: string) =>
  get<DeviceStatus>(`/api/v1/devices/${id}/status`)

export const listHttpServices = (deviceId: string) =>
  get<HttpService[]>(`/api/v1/devices/${deviceId}/http-services`)
